src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx

```diff
@@ -39,40 +38,31 @@ const AgencyBudgetLine = ({
         height: 0
     });
 
-    useEffect(() => {
-        if ((todaysDate >= xDomain[0]) && (todaysDate <= xDomain[1])) {
-            setShow(true);
-        }
-        else {
-            setShow(false);
-        }
-    }, [xScale, xDomain, todaysDate]);
-
     useEffect(() => {
         if (xScale && yScale && agencyBudget) {
-            const y = height - yScale(agencyBudget) - padding.top;
+            const y = height - yScale(agencyBudget) - padding.bottom;
             setLineData(
                 {
                     x1: padding.left,
-                    x2: show ? xScale(todaysDate) + padding.left : width - padding.left,
+                    x2: showTodayLineAndText ? xScale(todaysDate) + padding.left : width - padding.left,
                     y1: isNaN(y) ? 0 : y
                 }
             );
         }
-    }, [xScale, yScale, show]);
+    }, [xScale, yScale, showTodayLineAndText]);
 
     useEffect(() => {
         if (xScale && yScale && data.length) {
             setRectangleData(
                 {
                     x: padding.left,
-                    y: height - yScale(agencyBudget) - padding.top,
-                    width: show ? xScale(todaysDate) : width - padding.left - padding.right,
-                    height: height - yScale(data[data.length - 1].obligated) - padding.bottom - padding.top
+                    y: height - yScale(agencyBudget) - padding.bottom,
+                    width: showTodayLineAndText ? xScale(todaysDate) : width - padding.left - padding.right,
+                    height: height - yScale(Math.max(...data.map((x) => x.obligated))) - padding.bottom - padding.top
```

Comment:
Quick question on this change just for my own learning - are we getting the same height value using Math.max vs data.length - 1? Since math.max returns the largest value and data.length - 1 returns the last element. Just wanted to confirm the expected behavior here

Reply:
We pass the data from the container to the visualization sorted by period 👇  

`setData(addSubmissionEndDatesToBudgetaryResources(obligationsByPeriod, javaScriptSubmissionPeriods, fy).sort((a, b) => a.period - b.period));`

In theory, obligations should increase over time and the last period should be the greatest.  But we could have obligations that could be greater in earlier periods.